app/controllers/shipit/deploys_controller.rb

```diff
@@ -20,10 +20,15 @@ def show
     end
 
     def create
-      return redirect_to new_stack_deploy_path(@stack, sha: @until_commit.sha) if !params[:force] && @stack.active_task?
-
-      @deploy = @stack.trigger_deploy(@until_commit, current_user, env: deploy_params[:env])
+      @deploy = @stack.trigger_deploy(
+        @until_commit,
+        current_user,
+        env: deploy_params[:env],
+        force: params[:force].present?,
+      )
       respond_with(@deploy.stack, @deploy)
+    rescue Task::ConcurrentTaskRunning
+      redirect_to new_stack_deploy_path(@stack, sha: @until_commit.sha)
```

Comment:
It's already handled in the template. You got an orange warning asking you to confirm if you want to allow a concurrent deploy.